### ISSUE TITLE
Broken doctrine_orm_model_autocomplete in DatagridFilter

### ISSUE THREAD

**Issue by Nightbr** [sonata-project/SonataAdminBundle#4771]
Hey everyone, I noticed a little bug in sonataAdmin 3.25.x with doctrine_orm_model_autocomplete in DatagridFilter.

#### Sonata packages

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.25.1 3.25.1 The missing Symfony Admin Generator
sonata-project/block-bundle              3.5.0  3.5.0  Symfony SonataBlockBundle
sonata-project/cache                     2.0.0  2.0.0  Cache library
sonata-project/core-bundle               3.7.0  3.7.0  Symfony SonataCoreBundle
sonata-project/datagrid-bundle           2.2.1  2.2.1  Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.0.2  1.0.2  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.1.7  3.1.7  Symfony Sonata / Integrate Doctrine ORM i...
sonata-project/easy-extends-bundle       2.2.0  2.2.0  Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.7.1  1.7.1  Lightweight Exporter library
sonata-project/intl-bundle               2.3.2  2.3.2  Symfony SonataIntlBundle
sonata-project/media-bundle              3.8.0  3.8.0  Symfony SonataMediaBundle
sonata-project/notification-bundle       3.2.0  3.2.0  Symfony SonataNotificationBundle
```

#### Symfony packages

```
$ composer show --latest 'symfony/*'
symfony/monolog-bundle     v3.1.2  v3.1.2  Symfony MonologBundle
symfony/phpunit-bridge     v3.3.13 v3.3.13 Symfony PHPUnit Bridge
symfony/polyfill-apcu      v1.6.0  v1.6.0  Symfony polyfill backporting apcu_* functions to lowe...
symfony/polyfill-intl-icu  v1.6.0  v1.6.0  Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-mbstring  v1.6.0  v1.6.0  Symfony polyfill for the Mbstring extension
symfony/polyfill-php56     v1.6.0  v1.6.0  Symfony polyfill backporting some PHP 5.6+ features t...
symfony/polyfill-php70     v1.6.0  v1.6.0  Symfony polyfill backporting some PHP 7.0+ features t...
symfony/polyfill-php72     v1.6.0  v1.6.0  Symfony polyfill backporting some PHP 7.2+ features t...
symfony/polyfill-util      v1.6.0  v1.6.0  Symfony utilities for portability of PHP codes
symfony/security-acl       v3.0.0  v3.0.0  Symfony Security Component - ACL (Access Control List)
symfony/swiftmailer-bundle v2.6.7  v3.1.6  Symfony SwiftmailerBundle
symfony/symfony            v3.3.13 v3.3.13 The Symfony PHP framework

```

#### PHP version

```
$ php -v
PHP 7.1.4-1+deb.sury.org~xenial+1 (cli) (built: Apr 11 2017 22:12:32) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.4-1+deb.sury.org~xenial+1, Copyright (c) 1999-2017, by Zend Technologies
```

## Subject

Since sonata Admin 3.25.x all my list view throw 500 errors because of a problem in vendor/sonata-project/admin-bundle/src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig (line 40).

## Steps to reproduce

1. Install Sonata Admin 3.25.x
2. Create and admin class with a DatagridFilter with an orm_model_autocomplete filter:

For example:

```php
    /**
     * @param DatagridMapper $datagridMapper
     */
    protected function configureDatagridFilters(DatagridMapper $datagridMapper)
    {
        $datagridMapper
            ->add('id')
            ->add('title')
            ->add('category', 'doctrine_orm_choice', [], 'choice', array('choices' =>Publication::getCategoryChoices()))
            ->add('action', 'doctrine_orm_choice', [], 'choice', array('choices' =>Publication::getActionChoices()))
            ->add('place')
            ->add('enabled')
            ->add('isPublic')
            ->add('priority')
            ->add('author', 'doctrine_orm_model_autocomplete', array(), null, array(
                'property' => array('id', 'firstname', 'lastname', 'email'),
                'minimum_input_length' => 1,
            ))
            ->add('createdAt', 'doctrine_orm_datetime_range', array(
                'field_type' => 'sonata_type_datetime_range_picker',
            ))
        ;
    }

```

## Expected results

It should return the html page with 200 code, no errors and display the filters properly.

## Actual results

Error 500:

Stack trace:

Twig_Error_Runtime:
Impossible to access an attribute ("associationadmin") on a null variable.
```
  at vendor/sonata-project/admin-bundle/src/Resources/views/Form/Type/sonata_type_model_autocomplete.html.twig:40
  at twig_get_attribute(object(Twig_Environment), object(Twig_Source), null, 'associationadmin', array())
     (var/cache/dev/twig/56/5621b43cbd5276cc70c41dc4080a9ee424d1ab6e262f881d110af67ff729dfca.php:136)
  at __TwigTemplate_face685acaff48a89c6a2fc6d4e7266f5f4f25e043595a0a3b5ec09ce7e5df8d->doDisplay(array('value' => array(), 'attr' => array(), 'form' => object(FormView), 'id' => 'filter_user_value', 'name' => 'value', 'full_name' => 'filter[user][value]', 'disabled' => false, 'label' => null, 'label_format' => null, 'multipart' => false, 'block_prefixes' => array('form', 'sonata_type_model_autocomplete', '_filter_user_value'), 'unique_block_prefix' => '_filter_user_value', 'translation_domain' => null, 'cache_key' => '_filter_user_value_sonata_type_model_autocomplete', 'errors' => object(FormErrorIterator), 'valid' => true, 'data' => null, 'required' => false, 'size' => null, 'label_attr' => array(), 'compound' => false, 'method' => 'POST', 'action' => '', 'submitted' => true, 'sonata_admin_enabled' => false, 'sonata_help' => null, 'sonata_admin' => array('name' => null, 'admin' => null, 'value' => null, 'edit' => 'standard', 'inline' => 'natural', 'field_description' => null, 'block_name' => false, 'options' => array('html5_validate' => true, 'sort_admins' => false, 'confirm_exit' => true, 'use_select2' => true, 'use_icheck' => true, 'use_bootlint' => false, 'use_stickyforms' => true, 'pager_links' => null, 'form_type' => 'standard', 'dropdown_number_groups_per_colums' => 2, 'title_mode' => 'both', 'lock_protection' => false, 'enable_jms_di_extra_autoregistration' => true, 'javascripts' => array('bundles/sonatacore/vendor/jquery/dist/jquery.min.js', 'bundles/sonataadmin/vendor/jquery.scrollTo/jquery.scrollTo.min.js', 'bundles/sonatacore/vendor/moment/min/moment.min.js', 'bundles/sonataadmin/vendor/jqueryui/ui/minified/jquery-ui.min.js', 'bundles/sonataadmin/vendor/jqueryui/ui/minified/i18n/jquery-ui-i18n.min.js', 'bundles/sonatacore/vendor/bootstrap/dist/js/bootstrap.min.js', 'bundles/sonatacore/vendor/eonasdan-bootstrap-datetimepicker/build/js/bootstrap-datetimepicker.min.js', 'bundles/sonataadmin/vendor/jquery-form/jquery.form.js', 'bundles/sonataadmin/jquery/jquery.confirmExit.js', 'bundles/sonataadmin/vendor/x-editable/dist/bootstrap3-editable/js/bootstrap-editable.min.js', 'bundles/sonatacore/vendor/select2/select2.min.js', 'bundles/sonataadmin/vendor/admin-lte/dist/js/app.min.js', 'bundles/sonataadmin/vendor/iCheck/icheck.min.js', 'bundles/sonataadmin/vendor/slimScroll/jquery.slimscroll.min.js', 'bundles/sonataadmin/vendor/waypoints/lib/jquery.waypoints.min.js', 'bundles/sonataadmin/vendor/waypoints/lib/shortcuts/sticky.min.js', 'bundles/sonataadmin/vendor/readmore-js/readmore.min.js', 'bundles/sonataadmin/vendor/masonry/dist/masonry.pkgd.min.js', 'bundles/sonataadmin/Admin.js', 'bundles/sonataadmin/treeview.js'), 'stylesheets' => array('bundles/sonatacore/vendor/bootstrap/dist/css/bootstrap.min.css', 'bundles/sonatacore/vendor/components-font-awesome/css/font-awesome.min.css', 'bundles/sonatacore/vendor/ionicons/css/ionicons.min.css', 'bundles/sonataadmin/vendor/admin-lte/dist/css/AdminLTE.min.css', 'bundles/sonataadmin/vendor/admin-lte/dist/css/skins/skin-black.min.css', 'bundles/sonataadmin/vendor/iCheck/skins/square/blue.css', 'bundles/sonatacore/vendor/eonasdan-bootstrap-datetimepicker/build/css/bootstrap-datetimepicker.min.css', 'bundles/sonataadmin/vendor/jqueryui/themes/base/jquery-ui.css', 'bundles/sonatacore/vendor/select2/select2.css', 'bundles/sonatacore/vendor/select2-bootstrap-css/select2-bootstrap.min.css', 'bundles/sonataadmin/vendor/x-editable/dist/bootstrap3-editable/css/bootstrap-editable.css', 'bundles/sonataadmin/css/styles.css', 'bundles/sonataadmin/css/layout.css', 'bundles/sonataadmin/css/tree.css'), 'role_admin' => 'ROLE_SONATA_ADMIN', 'role_super_admin' => 'ROLE_SUPER_ADMIN')), 'horizontal_label_class' => '', 'horizontal_label_offset_class' => '', 'horizontal_input_wrapper_class' => '', 'admin_code' => 'sonata.admin.activity', 'placeholder' => '', 'multiple' => false, 'minimum_input_length' => 1, 'items_per_page' => 10, 'width' => '', 'url' => '', 'route' => array('name' => 'sonata_admin_retrieve_autocomplete_items', 'parameters' => array()), 'req_params' => array(), 'req_param_name_search' => 'q', 'req_param_name_page_number' => '_page', 'req_param_name_items_per_page' => '_per_page', 'quiet_millis' => 100, 'cache' => false, 'container_css_class' => '', 'dropdown_css_class' => '', 'dropdown_item_css_class' => '', 'dropdown_auto_width' => false, 'template' => 'SonataAdminBundle:Form/Type:sonata_type_model_autocomplete.html.twig', 'context' => 'filter', 'btn_add' => false, 'btn_catalogue' => 'SonataAdminBundle', 'wrap_fields_with_addons' => true, 'app' => object(AppVariable), 'sonata_media' => object(GlobalVariables), 'sonata_block' => object(GlobalVariables)), array('sonata_type_model_autocomplete_ajax_request_parameters' => array(object(__TwigTemplate_face685acaff48a89c6a2fc6d4e7266f5f4f25e043595a0a3b5ec09ce7e5df8d), 'block_sonata_type_model_autocomplete_ajax_request_parameters'), 'sonata_type_model_autocomplete_dropdown_item_format' => array(object(__TwigTemplate_face685acaff48a89c6a2fc6d4e7266f5f4f25e043595a0a3b5ec09ce7e5df8d), 'block_sonata_type_model_autocomplete_dropdown_item_format'), 'sonata_type_model_autocomplete_selection_format' => array(object(__TwigTemplate_face685acaff48a89c6a2fc6d4e7266f5f4f25e043595a0a3b5ec09ce7e5df8d), 'block_sonata_type_model_autocomplete_selection_format')))
     (vendor/twig/twig/lib/Twig/Template.php:389)
```

It seems to be related to your last feature: Add support for add button in sonata_type_model_autocomplete https://github.com/sonata-project/SonataAdminBundle/commit/946b6f2313c1222d1fac564f0ad770f20a34e585#diff-0cae783e3b2f6f22a7212e232c86ccfaR40

`sonata_admin.field_description` seems to be null and I think it's normal because it's a filter and you can't access to associationadmin from a null value. I think we should create a new condition to handle the case of filters.

Thanks in advance!

**Comment by greg0ire** [comment]
@psyray can you please help with this?

**Comment by psyray** [comment]
Hi,

@Nightbr Fixed with PR #4775 
Waiting to be merged.

**Comment by OskarStark** [comment]
Closed via #4775
